src/main.c

```diff
@@ -167,10 +169,10 @@ static bool parse_args(int argc, char **args)
     prog_args = &args[optind];
     opt_prog_name = prog_args[0];
     if (opt_prof_data) {
-        char *prog_name = malloc(strlen(opt_prog_name) - 11);
-        strncpy(prog_name, opt_prog_name + 8, strlen(opt_prog_name) - 12);
-        prof_out_file = malloc(strlen(opt_prog_name) + 1);
-        sprintf(prof_out_file, "./prof/%s.prof", prog_name);
+        char *prog_basename = basename(opt_prog_name);
+        prof_out_file = malloc(strlen(prog_basename) + 12 + 1);
+        assert(prof_out_file);
+        sprintf(prof_out_file, "./prof/%s.prof", prog_basename);
```

Comment:
The leading `prof/` directory name is not really making sense. Instead, let's place the profiling data in the same directory where the RV32 ELF file locates.

Reply:
I think you mean `build` directory

Reply:
I think putting it in the `build` directory is reasonable, we can distinguish the executable files and profiling data by filename extension.